### PR TITLE
feat(javascript): adds javascript test feature branch workflow

### PR DESCRIPTION
**Pull request by rob-telicent**
Shared workflow for testing javascript feature branches

### CODE REVIEW

File: .github/workflows/javascript-test-feature-branch.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+      - name: create .yarnrc file
+        run: |
+          npm config set "@telicent-io:registry" "https://TODO/"
+          npm config set "//TODO/:_authToken" ${{ secrets.NPMJS_AUTH_TOKEN }}
```
Comment (rob-telicent) [inline .github/workflows/javascript-test-feature-branch.yml:44]: What are the new registry URLs?

File: .github/workflows/javascript-test-feature-branch.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+      - name: create .yarnrc file
+        run: |
+          npm config set "@telicent-io:registry" "https://TODO/"
+          npm config set "//TODO/:_authToken" ${{ secrets.NPMJS_AUTH_TOKEN }} 
+          npm config set "@fortawesome:registry" https://npm.fontawesome.com/
+          npm config set  "//npm.fontawesome.com/:_authToken" ${{ secrets.FONT_AWESOME_AUTH_TOKEN }}
```
Comment (tatelicent) [inline .github/workflows/javascript-test-feature-branch.yml:46]: For the public stuff I dont think we will need this font awesome auth token and  config 
```suggestion
          
```

File: .github/workflows/javascript-test-feature-branch.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+      - name: create .yarnrc file
+        run: |
+          npm config set "@telicent-io:registry" "https://TODO/"
+          npm config set "//TODO/:_authToken" ${{ secrets.NPMJS_AUTH_TOKEN }} 
+          npm config set "@fortawesome:registry" https://npm.fontawesome.com/
```
Comment (tatelicent) [inline .github/workflows/javascript-test-feature-branch.yml:45]:
```suggestion
          
```

File: .github/workflows/javascript-test-feature-branch.yml (hunk 2 above)
Comment (tatelicent) [reply .github/workflows/javascript-test-feature-branch.yml:46]: @davec504 is this correct?

File: .github/workflows/javascript-test-feature-branch.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        id: yarn-cache-dir-path
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-node_modules-${{ hashFiles('**/yarn.lock') }}-${{ steps.node.outputs.version }}
+      - name: create .yarnrc file
+        run: |
+          npm config set "@telicent-io:registry" "https://www.npmjs.com/"
```
Comment (tatelicent) [inline .github/workflows/javascript-test-feature-branch.yml:43]:
```suggestion
          npm config set "@telicent-oss:registry" "https://www.npmjs.com/"
```

**Review by tatelicent — APPROVED** [review]

**Review by davec504 — APPROVED** [review]

**Pull request merged**